app/src/main/java/com/xmartlabs/template/ui/BaseFragment.java

```diff
@@ -0,0 +1,140 @@
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.annimon.stream.Optional;
+import com.hannesdorfmann.fragmentargs.FragmentArgs;
+import com.trello.rxlifecycle.components.support.RxFragment;
+import com.xmartlabs.template.BaseProjectApplication;
+import com.xmartlabs.template.ui.common.BaseProgressDialog;
+
+import butterknife.ButterKnife;
+import butterknife.Unbinder;
+
+/**
+ * Base Fragment implementation with the following functionality:
+ * <ul>
+ *   <li>Inflate the view given a layout resource</li>
+ *   <li>Bind the view layout elements with ButterKnife</li>
+ *   <li>Ability to show/hide a progress dialog of your choosing (providing it extends from <code>BaseProgressDialog</code></li>
+ *   <li>If the activity that holds this Fragment extends from <code>BaseAppCompatActivity</code>, allows the instance
+ *       to be removed</li>
+ *   <li>Ability to remove itself from parent fragment</li>
+ *   <li>Proper cleanup on detach/destroy</li>
+ * </ul>
+ */
+public abstract class BaseFragment extends RxFragment {
+  private Unbinder unbinder;
+  @Nullable
+  private BaseProgressDialog progressDialog;
+
+  /**
+   * Used to inflate the view layout/elements
+   * @return the layout resource from which to inflate the view
+   */
+  @LayoutRes
+  protected abstract int getLayoutResId();
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    FragmentArgs.inject(this);
+    BaseProjectApplication.getContext().inject(this);
+  }
+
+  @NonNull
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    View view = inflater.inflate(getLayoutResId(), container, false);
+    unbinder = ButterKnife.bind(this, view);
+    return view;
+  }
+
+  @Override
+  public void onAttach(Activity activity) {
+    super.onAttach(activity);
+    progressDialog = createProgressDialog();
+  }
+
+  /**
+   * To be used to show/hide a progress dialog.
+   * The dialog must extend from BaseProgressDialog
+   * @return the BaseProgressDialog instance to be shown upon request
+   */
+  @Nullable
+  protected BaseProgressDialog createProgressDialog() {
+    return null;
+  }
+
+  @Override
+  public void onDetach() {
+    super.onDetach();
+    Optional.ofNullable(progressDialog)
+        .ifPresent(BaseProgressDialog::dismiss);
+    progressDialog = null;
+  }
+
+  @Override
+  public void onDestroyView() {
+    super.onDestroyView();
+    unbinder.unbind();
+  }
+
+  /**
+   * Show a simple alert with an ok button
+   * @param stringResId the message to be shown in the alert
+   */
+  protected void showAlertError(int stringResId) {
+    new AlertDialog.Builder(getContext())
```

Comment:
Yup, added here 
[https://trello.com/c/etigX9Md/19-abstract-the-alertdialog-in-the-base-fragment-as-we-did-with-the-dialog](https://trello.com/c/etigX9Md/19-abstract-the-alertdialog-in-the-base-fragment-as-we-did-with-the-dialog)